src/crimson/net/Protocol.h

```diff
@@ -85,13 +86,16 @@ class Protocol {
  private:
   bool closed = false;
   // become valid only after closed == true
-  seastar::shared_future<> close_ready;
+  seastar::shared_promise<> close_ready;
```

Comment:
Why need to change to `shared_promise<>`?
The original `seastar::shared_future<> close_ready` should be already working.